aws/resource_aws_lb.go

```diff
@@ -687,3 +693,50 @@ func flattenAwsLbResource(d *schema.ResourceData, meta interface{}, lb *elbv2.Lo
 
 	return nil
 }
+
+// Load balancers of type 'network' cannot have their subnets updated at
+// this time. If the type is 'network' and subnets have changed, mark the
+// diff as a ForceNew operation
+func customizeDiffNLBSubnets(diff *schema.ResourceDiff, v interface{}) error {
+	// The current criteria for determining if the operation should be ForceNew:
+	// - lb of type "network"
+	// - existing resource (id is not "")
+	// - there are actual changes to be made in the subnets
+	//
+	// Any other combination should be treated as normal. At this time, subnet
+	// handling is the only known difference between Network Load Balancers and
+	// Application Load Balancers, so the logic below is simple individual checks.
+	// If other differences arise we'll want to refactor to check other
+	// conditions in combinations, but for now all we handle is subnets
+	lbType := diff.Get("load_balancer_type").(string)
+	if "network" != lbType {
+		return nil
+	}
+
+	if "" == diff.Id() {
```

Comment:
nitpick, but `diff.Id() == ""` would make more sense. 😉

Reply:
Old defensive habit from my Ruby days, trying to save myself from typos like 

```
if variable = "something" {
  //stuff 
}
``` 

when I meant 

```
if variable == "something" {
  //stuff
}
```